services/queue/src/utils.js

```diff
@@ -109,3 +110,35 @@ const artifactUtils = {
 };
 
 exports.artifactUtils = artifactUtils;
+
+/**
+ * Split a taskQueueId into its deprecated provisionerId/workerType components.
+ */
+const splitTaskQueueId = taskQueueId => {
+  const split = taskQueueId.split('/');
+  assert.equal(split.length, 2, `invalid taskQueueId ${taskQueueId}`);
+  return { provisionerId: split[0], workerType: split[1] };
+};
+exports.splitTaskQueueId = splitTaskQueueId;
+
+/**
+ * Join a provisionerId and workerType to make a taskQueueId
+ */
+const joinTaskQueueId = (provisionerId, workerType) => {
+  assert(typeof provisionerId === 'string', 'provisionerId omitted');
+  assert(typeof workerType === 'string', 'workerType omitted');
+  assert(provisionerId.indexOf('/') === -1, 'provisionerId cannot contain `/`');
+  return `${provisionerId}/${workerType}`;
+};
+exports.joinTaskQueueId = joinTaskQueueId;
+
+/**
+ * Add the provisionerId, workerType to an object that has a
+ * taskQueueId to maintain public interface compatibility
+ */
+const addSplitFields = (obj) => {
+  const { provisionerId, workerType } = splitTaskQueueId(obj.taskQueueId);
+  obj.provisionerId = provisionerId;
+  obj.workerType = workerType;
```

Comment:
Maybe rename this to `useSplitFields` and have it also `delete obj.taskQueueId`?  And perhaps a mirror-image `fromSplitFields` that takes an API input with `provisionerId` and `workerType` and replaces those with `taskQueueId`?

Reply:
Looks like a good idea!